From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Mon, 12 Mar 2018 20:31:44 +0100
Subject: Add Qwant search engine

Specify image URL in upstream engine definition
Make search engine selectable in default list
---
 .../search_engines/prepopulated_engines.json  |   1 +
 .../template_url_prepopulate_data.cc          | 216 +++++++++---------
 2 files changed, 109 insertions(+), 108 deletions(-)

diff --git a/components/search_engines/prepopulated_engines.json b/components/search_engines/prepopulated_engines.json
--- a/components/search_engines/prepopulated_engines.json
+++ b/components/search_engines/prepopulated_engines.json
@@ -183,6 +183,7 @@
       "favicon_url": "https://www.qwant.com/favicon.ico",
       "search_url": "https://www.qwant.com/?q={searchTerms}&client=opensearch",
       "suggest_url": "https://api.qwant.com/api/suggest/?q={searchTerms}&client=opensearch",
+      "image_url": "https://www.qwant.com/?t=images&q={searchTerms}",
       "type": "SEARCH_ENGINE_QWANT",
       "id": 94
     },
diff --git a/components/search_engines/template_url_prepopulate_data.cc b/components/search_engines/template_url_prepopulate_data.cc
--- a/components/search_engines/template_url_prepopulate_data.cc
+++ b/components/search_engines/template_url_prepopulate_data.cc
@@ -32,7 +32,7 @@ namespace {
 
 // Default (for countries with no better engine set)
 const PrepopulatedEngine* const engines_default[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -42,7 +42,7 @@ const PrepopulatedEngine* const engines_default[] = {
 // Engine selection by country ------------------------------------------------
 // United Arab Emirates
 const PrepopulatedEngine* const engines_AE[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -50,7 +50,7 @@ const PrepopulatedEngine* const engines_AE[] = {
 
 // Albania
 const PrepopulatedEngine* const engines_AL[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo,
     &bing,
     &ask,
@@ -58,7 +58,7 @@ const PrepopulatedEngine* const engines_AL[] = {
 
 // Argentina
 const PrepopulatedEngine* const engines_AR[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo_ar,
     &duckduckgo, &duckduckgo_light,
@@ -66,7 +66,7 @@ const PrepopulatedEngine* const engines_AR[] = {
 
 // Austria
 const PrepopulatedEngine* const engines_AT[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo_at,
     &duckduckgo, &duckduckgo_light,
@@ -74,7 +74,7 @@ const PrepopulatedEngine* const engines_AT[] = {
 
 // Australia
 const PrepopulatedEngine* const engines_AU[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo_au,
     &duckduckgo, &duckduckgo_light,
@@ -82,7 +82,7 @@ const PrepopulatedEngine* const engines_AU[] = {
 
 // Bosnia and Herzegovina
 const PrepopulatedEngine* const engines_BA[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo,
     &bing,
     &ask,
@@ -90,7 +90,7 @@ const PrepopulatedEngine* const engines_BA[] = {
 
 // Belgium
 const PrepopulatedEngine* const engines_BE[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -98,7 +98,7 @@ const PrepopulatedEngine* const engines_BE[] = {
 
 // Bulgaria
 const PrepopulatedEngine* const engines_BG[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &yandex_ru,
@@ -106,7 +106,7 @@ const PrepopulatedEngine* const engines_BG[] = {
 
 // Bahrain
 const PrepopulatedEngine* const engines_BH[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &ask,
@@ -114,7 +114,7 @@ const PrepopulatedEngine* const engines_BH[] = {
 
 // Burundi
 const PrepopulatedEngine* const engines_BI[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &yandex_ru,
@@ -122,7 +122,7 @@ const PrepopulatedEngine* const engines_BI[] = {
 
 // Brunei
 const PrepopulatedEngine* const engines_BN[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -130,7 +130,7 @@ const PrepopulatedEngine* const engines_BN[] = {
 
 // Bolivia
 const PrepopulatedEngine* const engines_BO[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -138,7 +138,7 @@ const PrepopulatedEngine* const engines_BO[] = {
 
 // Brazil
 const PrepopulatedEngine* const engines_BR[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo_br,
     &duckduckgo, &duckduckgo_light,
@@ -146,7 +146,7 @@ const PrepopulatedEngine* const engines_BR[] = {
 
 // Belarus
 const PrepopulatedEngine* const engines_BY[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yandex_by,
     &mail_ru,
     &bing,
@@ -154,7 +154,7 @@ const PrepopulatedEngine* const engines_BY[] = {
 
 // Belize
 const PrepopulatedEngine* const engines_BZ[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -163,12 +163,12 @@ const PrepopulatedEngine* const engines_BZ[] = {
 
 // Canada
 const PrepopulatedEngine* const engines_CA[] = {
-    &googleen, &google, &bing, &yahoo_ca, &yahoo_qc, &duckduckgo, &duckduckgo_light,
+    &googleen, &google, &qwant, &bing, &yahoo_ca, &yahoo_qc, &duckduckgo, &duckduckgo_light,
 };
 
 // Switzerland
 const PrepopulatedEngine* const engines_CH[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &duckduckgo,
     &duckduckgo_light,
@@ -177,7 +177,7 @@ const PrepopulatedEngine* const engines_CH[] = {
 
 // Chile
 const PrepopulatedEngine* const engines_CL[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo_cl,
     &duckduckgo, &duckduckgo_light,
@@ -185,12 +185,12 @@ const PrepopulatedEngine* const engines_CL[] = {
 
 // China
 const PrepopulatedEngine* const engines_CN[] = {
-    &baidu, &so_360, &googleen, &google, &sogou, &bing,
+    &baidu, &so_360, &googleen, &google, &qwant, &sogou, &bing,
 };
 
 // Colombia
 const PrepopulatedEngine* const engines_CO[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo_co,
     &bing,
     &duckduckgo, &duckduckgo_light,
@@ -198,7 +198,7 @@ const PrepopulatedEngine* const engines_CO[] = {
 
 // Costa Rica
 const PrepopulatedEngine* const engines_CR[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -206,7 +206,7 @@ const PrepopulatedEngine* const engines_CR[] = {
 
 // Czech Republic
 const PrepopulatedEngine* const engines_CZ[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &seznam_cz,
     &bing,
     &yahoo,
@@ -214,7 +214,7 @@ const PrepopulatedEngine* const engines_CZ[] = {
 
 // Germany
 const PrepopulatedEngine* const engines_DE[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo_de,
     &duckduckgo, &duckduckgo_light,
@@ -222,7 +222,7 @@ const PrepopulatedEngine* const engines_DE[] = {
 
 // Denmark
 const PrepopulatedEngine* const engines_DK[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo_dk,
     &duckduckgo, &duckduckgo_light,
@@ -230,7 +230,7 @@ const PrepopulatedEngine* const engines_DK[] = {
 
 // Dominican Republic
 const PrepopulatedEngine* const engines_DO[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -238,7 +238,7 @@ const PrepopulatedEngine* const engines_DO[] = {
 
 // Algeria
 const PrepopulatedEngine* const engines_DZ[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo,
     &bing,
     &yandex_ru,
@@ -246,7 +246,7 @@ const PrepopulatedEngine* const engines_DZ[] = {
 
 // Ecuador
 const PrepopulatedEngine* const engines_EC[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -254,7 +254,7 @@ const PrepopulatedEngine* const engines_EC[] = {
 
 // Estonia
 const PrepopulatedEngine* const engines_EE[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yandex_ru,
     &neti,
@@ -262,7 +262,7 @@ const PrepopulatedEngine* const engines_EE[] = {
 
 // Egypt
 const PrepopulatedEngine* const engines_EG[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo,
     &bing,
     &yandex_ru,
@@ -270,7 +270,7 @@ const PrepopulatedEngine* const engines_EG[] = {
 
 // Spain
 const PrepopulatedEngine* const engines_ES[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo_es,
     &duckduckgo, &duckduckgo_light,
@@ -278,7 +278,7 @@ const PrepopulatedEngine* const engines_ES[] = {
 
 // Finland
 const PrepopulatedEngine* const engines_FI[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo_fi,
     &duckduckgo, &duckduckgo_light,
@@ -286,7 +286,7 @@ const PrepopulatedEngine* const engines_FI[] = {
 
 // Faroe Islands
 const PrepopulatedEngine* const engines_FO[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -294,7 +294,7 @@ const PrepopulatedEngine* const engines_FO[] = {
 
 // France
 const PrepopulatedEngine* const engines_FR[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo_fr,
     &qwant,
@@ -302,7 +302,7 @@ const PrepopulatedEngine* const engines_FR[] = {
 
 // United Kingdom
 const PrepopulatedEngine* const engines_GB[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo_uk,
     &duckduckgo, &duckduckgo_light,
@@ -310,7 +310,7 @@ const PrepopulatedEngine* const engines_GB[] = {
 
 // Greece
 const PrepopulatedEngine* const engines_GR[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -318,7 +318,7 @@ const PrepopulatedEngine* const engines_GR[] = {
 
 // Guatemala
 const PrepopulatedEngine* const engines_GT[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -326,7 +326,7 @@ const PrepopulatedEngine* const engines_GT[] = {
 
 // Hong Kong
 const PrepopulatedEngine* const engines_HK[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo_hk,
     &bing,
     &baidu,
@@ -334,7 +334,7 @@ const PrepopulatedEngine* const engines_HK[] = {
 
 // Honduras
 const PrepopulatedEngine* const engines_HN[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -342,7 +342,7 @@ const PrepopulatedEngine* const engines_HN[] = {
 
 // Croatia
 const PrepopulatedEngine* const engines_HR[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -350,7 +350,7 @@ const PrepopulatedEngine* const engines_HR[] = {
 
 // Hungary
 const PrepopulatedEngine* const engines_HU[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo,
     &bing,
     &duckduckgo, &duckduckgo_light,
@@ -358,7 +358,7 @@ const PrepopulatedEngine* const engines_HU[] = {
 
 // Indonesia
 const PrepopulatedEngine* const engines_ID[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo_id,
     &bing,
     &duckduckgo, &duckduckgo_light,
@@ -366,7 +366,7 @@ const PrepopulatedEngine* const engines_ID[] = {
 
 // Ireland
 const PrepopulatedEngine* const engines_IE[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -374,7 +374,7 @@ const PrepopulatedEngine* const engines_IE[] = {
 
 // Israel
 const PrepopulatedEngine* const engines_IL[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &yandex_ru,
@@ -382,7 +382,7 @@ const PrepopulatedEngine* const engines_IL[] = {
 
 // India
 const PrepopulatedEngine* const engines_IN[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo_in,
     &bing,
     &duckduckgo, &duckduckgo_light,
@@ -390,7 +390,7 @@ const PrepopulatedEngine* const engines_IN[] = {
 
 // Iraq
 const PrepopulatedEngine* const engines_IQ[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo,
     &bing,
     &yandex_ru,
@@ -398,14 +398,14 @@ const PrepopulatedEngine* const engines_IQ[] = {
 
 // Iran
 const PrepopulatedEngine* const engines_IR[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
 };
 
 // Iceland
 const PrepopulatedEngine* const engines_IS[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -413,7 +413,7 @@ const PrepopulatedEngine* const engines_IS[] = {
 
 // Italy
 const PrepopulatedEngine* const engines_IT[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -421,7 +421,7 @@ const PrepopulatedEngine* const engines_IT[] = {
 
 // Jamaica
 const PrepopulatedEngine* const engines_JM[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -429,7 +429,7 @@ const PrepopulatedEngine* const engines_JM[] = {
 
 // Jordan
 const PrepopulatedEngine* const engines_JO[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo,
     &bing,
     &yandex_ru,
@@ -437,7 +437,7 @@ const PrepopulatedEngine* const engines_JO[] = {
 
 // Japan
 const PrepopulatedEngine* const engines_JP[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo_jp,
     &bing,
     &baidu,
@@ -445,7 +445,7 @@ const PrepopulatedEngine* const engines_JP[] = {
 
 // Kenya
 const PrepopulatedEngine* const engines_KE[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo,
     &bing,
     &duckduckgo, &duckduckgo_light,
@@ -453,7 +453,7 @@ const PrepopulatedEngine* const engines_KE[] = {
 
 // South Korea
 const PrepopulatedEngine* const engines_KR[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &naver,
     &daum,
     &bing,
@@ -461,7 +461,7 @@ const PrepopulatedEngine* const engines_KR[] = {
 
 // Kuwait
 const PrepopulatedEngine* const engines_KW[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &ask,
@@ -469,7 +469,7 @@ const PrepopulatedEngine* const engines_KW[] = {
 
 // Kazakhstan
 const PrepopulatedEngine* const engines_KZ[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yandex_kz,
     &mail_ru,
     &bing,
@@ -477,7 +477,7 @@ const PrepopulatedEngine* const engines_KZ[] = {
 
 // Lebanon
 const PrepopulatedEngine* const engines_LB[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -486,7 +486,7 @@ const PrepopulatedEngine* const engines_LB[] = {
 
 // Liechtenstein
 const PrepopulatedEngine* const engines_LI[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &duckduckgo,
     &duckduckgo_light,
@@ -495,7 +495,7 @@ const PrepopulatedEngine* const engines_LI[] = {
 
 // Lithuania
 const PrepopulatedEngine* const engines_LT[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &yandex_ru,
@@ -503,7 +503,7 @@ const PrepopulatedEngine* const engines_LT[] = {
 
 // Luxembourg
 const PrepopulatedEngine* const engines_LU[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &duckduckgo, &duckduckgo_light,
     &yahoo,
@@ -511,7 +511,7 @@ const PrepopulatedEngine* const engines_LU[] = {
 
 // Latvia
 const PrepopulatedEngine* const engines_LV[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yandex_ru,
     &yahoo,
@@ -519,7 +519,7 @@ const PrepopulatedEngine* const engines_LV[] = {
 
 // Libya
 const PrepopulatedEngine* const engines_LY[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo,
     &bing,
     &yandex_com,
@@ -527,7 +527,7 @@ const PrepopulatedEngine* const engines_LY[] = {
 
 // Morocco
 const PrepopulatedEngine* const engines_MA[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo,
     &bing,
     &yandex_ru,
@@ -535,7 +535,7 @@ const PrepopulatedEngine* const engines_MA[] = {
 
 // Monaco
 const PrepopulatedEngine* const engines_MC[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -543,7 +543,7 @@ const PrepopulatedEngine* const engines_MC[] = {
 
 // Moldova
 const PrepopulatedEngine* const engines_MD[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yandex_ru,
     &mail_ru,
     &duckduckgo, &duckduckgo_light,
@@ -551,7 +551,7 @@ const PrepopulatedEngine* const engines_MD[] = {
 
 // Montenegro
 const PrepopulatedEngine* const engines_ME[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo,
     &bing,
     &yandex_ru,
@@ -559,7 +559,7 @@ const PrepopulatedEngine* const engines_ME[] = {
 
 // Macedonia
 const PrepopulatedEngine* const engines_MK[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo,
     &bing,
     &duckduckgo, &duckduckgo_light,
@@ -567,7 +567,7 @@ const PrepopulatedEngine* const engines_MK[] = {
 
 // Mexico
 const PrepopulatedEngine* const engines_MX[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo_mx,
     &duckduckgo, &duckduckgo_light,
@@ -575,7 +575,7 @@ const PrepopulatedEngine* const engines_MX[] = {
 
 // Malaysia
 const PrepopulatedEngine* const engines_MY[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo_my,
     &bing,
     &baidu,
@@ -583,7 +583,7 @@ const PrepopulatedEngine* const engines_MY[] = {
 
 // Nicaragua
 const PrepopulatedEngine* const engines_NI[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -591,7 +591,7 @@ const PrepopulatedEngine* const engines_NI[] = {
 
 // Netherlands
 const PrepopulatedEngine* const engines_NL[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo_nl,
     &duckduckgo, &duckduckgo_light,
@@ -599,7 +599,7 @@ const PrepopulatedEngine* const engines_NL[] = {
 
 // Norway
 const PrepopulatedEngine* const engines_NO[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo,
@@ -608,7 +608,7 @@ const PrepopulatedEngine* const engines_NO[] = {
 
 // New Zealand
 const PrepopulatedEngine* const engines_NZ[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &duckduckgo, &duckduckgo_light,
     &yahoo_nz,
@@ -616,7 +616,7 @@ const PrepopulatedEngine* const engines_NZ[] = {
 
 // Oman
 const PrepopulatedEngine* const engines_OM[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &ask,
@@ -624,7 +624,7 @@ const PrepopulatedEngine* const engines_OM[] = {
 
 // Panama
 const PrepopulatedEngine* const engines_PA[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -632,7 +632,7 @@ const PrepopulatedEngine* const engines_PA[] = {
 
 // Peru
 const PrepopulatedEngine* const engines_PE[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo_pe,
     &bing,
     &duckduckgo, &duckduckgo_light,
@@ -640,7 +640,7 @@ const PrepopulatedEngine* const engines_PE[] = {
 
 // Philippines
 const PrepopulatedEngine* const engines_PH[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo_ph,
     &bing,
     &duckduckgo, &duckduckgo_light,
@@ -648,7 +648,7 @@ const PrepopulatedEngine* const engines_PH[] = {
 
 // Pakistan
 const PrepopulatedEngine* const engines_PK[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo,
     &bing,
     &baidu,
@@ -656,7 +656,7 @@ const PrepopulatedEngine* const engines_PK[] = {
 
 // Poland
 const PrepopulatedEngine* const engines_PL[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -664,7 +664,7 @@ const PrepopulatedEngine* const engines_PL[] = {
 
 // Puerto Rico
 const PrepopulatedEngine* const engines_PR[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -672,7 +672,7 @@ const PrepopulatedEngine* const engines_PR[] = {
 
 // Portugal
 const PrepopulatedEngine* const engines_PT[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -680,7 +680,7 @@ const PrepopulatedEngine* const engines_PT[] = {
 
 // Paraguay
 const PrepopulatedEngine* const engines_PY[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -688,7 +688,7 @@ const PrepopulatedEngine* const engines_PY[] = {
 
 // Qatar
 const PrepopulatedEngine* const engines_QA[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &ask,
@@ -696,7 +696,7 @@ const PrepopulatedEngine* const engines_QA[] = {
 
 // Romania
 const PrepopulatedEngine* const engines_RO[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -704,7 +704,7 @@ const PrepopulatedEngine* const engines_RO[] = {
 
 // Serbia
 const PrepopulatedEngine* const engines_RS[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo,
     &bing,
     &duckduckgo, &duckduckgo_light,
@@ -713,14 +713,14 @@ const PrepopulatedEngine* const engines_RS[] = {
 // Russia
 const PrepopulatedEngine* const engines_RU[] = {
     &yandex_ru,
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &mail_ru,
     &bing,
 };
 
 // Rwanda
 const PrepopulatedEngine* const engines_RW[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &yandex_ru,
@@ -728,7 +728,7 @@ const PrepopulatedEngine* const engines_RW[] = {
 
 // Saudi Arabia
 const PrepopulatedEngine* const engines_SA[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &yandex_com,
@@ -736,7 +736,7 @@ const PrepopulatedEngine* const engines_SA[] = {
 
 // Sweden
 const PrepopulatedEngine* const engines_SE[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo_se,
     &duckduckgo, &duckduckgo_light,
@@ -744,7 +744,7 @@ const PrepopulatedEngine* const engines_SE[] = {
 
 // Singapore
 const PrepopulatedEngine* const engines_SG[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo_sg,
     &yandex_com,
@@ -752,7 +752,7 @@ const PrepopulatedEngine* const engines_SG[] = {
 
 // Slovenia
 const PrepopulatedEngine* const engines_SI[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -760,7 +760,7 @@ const PrepopulatedEngine* const engines_SI[] = {
 
 // Slovakia
 const PrepopulatedEngine* const engines_SK[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -768,7 +768,7 @@ const PrepopulatedEngine* const engines_SK[] = {
 
 // El Salvador
 const PrepopulatedEngine* const engines_SV[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -776,7 +776,7 @@ const PrepopulatedEngine* const engines_SV[] = {
 
 // Syria
 const PrepopulatedEngine* const engines_SY[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &yandex_ru,
@@ -784,7 +784,7 @@ const PrepopulatedEngine* const engines_SY[] = {
 
 // Thailand
 const PrepopulatedEngine* const engines_TH[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo_th,
     &bing,
     &ask,
@@ -792,7 +792,7 @@ const PrepopulatedEngine* const engines_TH[] = {
 
 // Tunisia
 const PrepopulatedEngine* const engines_TN[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo,
     &bing,
     &yandex_ru,
@@ -800,7 +800,7 @@ const PrepopulatedEngine* const engines_TN[] = {
 
 // Turkey
 const PrepopulatedEngine* const engines_TR[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yandex_tr,
     &yahoo_tr,
     &bing,
@@ -808,7 +808,7 @@ const PrepopulatedEngine* const engines_TR[] = {
 
 // Trinidad and Tobago
 const PrepopulatedEngine* const engines_TT[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -816,7 +816,7 @@ const PrepopulatedEngine* const engines_TT[] = {
 
 // Taiwan
 const PrepopulatedEngine* const engines_TW[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo_tw,
     &bing,
     &baidu,
@@ -824,7 +824,7 @@ const PrepopulatedEngine* const engines_TW[] = {
 
 // Tanzania
 const PrepopulatedEngine* const engines_TZ[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &yandex_ru,
@@ -832,7 +832,7 @@ const PrepopulatedEngine* const engines_TZ[] = {
 
 // Ukraine
 const PrepopulatedEngine* const engines_UA[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yandex_ua,
     &mail_ru,
     &bing,
@@ -840,7 +840,7 @@ const PrepopulatedEngine* const engines_UA[] = {
 
 // United States
 const PrepopulatedEngine* const engines_US[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -848,7 +848,7 @@ const PrepopulatedEngine* const engines_US[] = {
 
 // Uruguay
 const PrepopulatedEngine* const engines_UY[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -856,7 +856,7 @@ const PrepopulatedEngine* const engines_UY[] = {
 
 // Venezuela
 const PrepopulatedEngine* const engines_VE[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &yahoo_ve,
     &bing,
     &duckduckgo, &duckduckgo_light,
@@ -864,7 +864,7 @@ const PrepopulatedEngine* const engines_VE[] = {
 
 // Vietnam
 const PrepopulatedEngine* const engines_VN[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &coccoc,
     &bing,
     &yahoo_vn,
@@ -872,7 +872,7 @@ const PrepopulatedEngine* const engines_VN[] = {
 
 // Yemen
 const PrepopulatedEngine* const engines_YE[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &baidu,
@@ -880,7 +880,7 @@ const PrepopulatedEngine* const engines_YE[] = {
 
 // South Africa
 const PrepopulatedEngine* const engines_ZA[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &duckduckgo, &duckduckgo_light,
@@ -888,7 +888,7 @@ const PrepopulatedEngine* const engines_ZA[] = {
 
 // Zimbabwe
 const PrepopulatedEngine* const engines_ZW[] = {
-    &googleen, &google,
+    &googleen, &google, &qwant,
     &bing,
     &yahoo,
     &baidu,
-- 
2.17.1

